src/State/Hub.php

```diff
@@ -230,22 +231,35 @@ private function getStackTop(): Layer
     public function startTransaction(TransactionContext $context): Transaction
     {
         $client = $this->getClient();
+        $sampleRate = null;
+
+        if (null !== $client) {
+            $sampler = $client->getOptions()->getTracesSampler();
+
+            if (null !== $sampler) {
+                if (\is_callable($sampler)) {
```

Comment:
Once we typehint the getter and the setter this check can be replaced with a simpler `null !== $sampler`